include/ocpp/common/types.hpp

```diff
@@ -33,6 +33,23 @@ struct Message {
     virtual std::string get_type() const = 0;
 };
 
+/// \brief Exception used when DateTime class is initialized by invalid timepoint string.
+class TimePointParseException : public std::exception {
+public:
+    explicit TimePointParseException(const std::string& timepoint_str) :
+        msg{"Timepoint string parsing failed. Could not convert: \"" + timepoint_str + "\" into DateTime."} {
+    }
+
+    ~TimePointParseException() noexcept override = default;
```

Comment:
Don't think we need to explicitly define this if it is a default destructor

Reply:
Just copied from other exception classes here, but indeed, the compiler will generate the same destructor for us